R/bundleFiles.R

```diff
@@ -127,14 +87,25 @@ bundleFiles <- function(appDir) {
   listBundleFiles(appDir)$contents
 }
 
+explodeFiles <- function(dir, files) {
+  files <- files[file.exists(file.path(dir, files))]
+  recursiveBundleFiles(dir, contents = files, ignoreFiles = FALSE)$contents
+}
+
 recursiveBundleFiles <- function(dir,
+                                 contents = NULL,
                                  rootDir = dir,
                                  depth = 0,
                                  totalFiles = 0,
-                                 totalSize = 0) {
+                                 totalSize = 0,
+                                 ignoreFiles = TRUE) {
   # generate a list of files at this level
-  contents <- list.files(dir, all.files = TRUE, no.. = TRUE, include.dirs = TRUE)
-  contents <- ignoreBundleFiles(dir, contents, depth = depth)
+  if (is.null(contents)) {
```

Comment:
Question: Does this recursion get confused if some of the incoming `contents` are complex paths rather than paths at the root level? The depth calculation could be off with something like: `contents = c("index.Rmd", "images/logos/logo.png")`

Reply:
`depth` is only needed for ignoring files and will go away in #700, so we can dodge that question 😄